forgerock-auth/src/main/java/org/forgerock/android/auth/AppAuthFragment2.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+package org.forgerock.android.auth
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.annotation.RestrictTo
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+import androidx.fragment.app.FragmentManager
+import kotlinx.coroutines.flow.MutableStateFlow
+import net.openid.appauth.AuthorizationResponse
+import org.forgerock.android.auth.centralize.BrowserLauncher
+import org.forgerock.android.auth.centralize.Launcher
+
+/**
+ * Headless Fragment to receive callback result from AppAuth library
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY)
+class AppAuthFragment2 : Fragment() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val state: MutableStateFlow<Intent?> = MutableStateFlow(null)
+        val delegate =
+            registerForActivityResult(AuthorizeContract()) {
+                state.value = it
+            }
+
+        val pending = savedInstanceState?.getBoolean("pending", false) ?: false
```

Comment:
just put this in val and reuse the key -  private val savedInstanceStateKey = "pending"

Reply:
Updated.